### DISCUSSION TITLE
Send Async Response while my controller api method being Void ( javax @AsyncResponse style)

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#10563] [Q&A]
Hi, 
Is there a way I can keep my api method void, but use kind of a handler I can use later downstream in my code to send the response.
Example:
Usually I would have following. 

```
    public Publisher< MyResponse.class > post() {
        return Mono.from(client.retrieve(url), MyResponse.class));
    }

```

Instead expected following. Here mysvc.invoke() method will call say a backend api in an async manner and send a Publisher<MyResponse> to the client.

```
    public void post() {
        mysvc.invoke()
    }

```

Main reason is we have some apis uses javax AsyncResponse where you can keep a   void api method and send response later. 
This is to support similar behavior for such written code.

**Comment by subash89** [comment]
Appreciate any input.

**Comment by yawkat** [comment]
i mean, you can use Mono.create(sink -> ...)
